File: frontend/src/types/clinic.js
// Clinic types
export const Clinic = {
  clinicId: '',
  name: '',
  address: '',
  phone: '',
  description: '',
  createdAt: '',
  clinicImages: [],
};

export const ClinicCreateDTO = {
  name: '',
  phone: '',
  address: '',
  description: '',
  clinicImages: [],
};

export const ClinicUpdateDTO = {
  name: '',
  address: '',
  phone: '',
  clinicImamges: [],
  description: '',
};

export const ClinicInfoDTO = {
  clinicId: '',
  name: '',
  address: '',
};

export const ClinicDetailInfoDTO = {
  clinicId: '',
  name: '',
  address: '',
  description: '',
  phone: '',
  createdAt: '',
  reviews: [],
  clinicImages: [],
};

File: backend/src/main/java/com/project2/demo/Models/Clinic.java
package com.project2.demo.Models;
import java.time.LocalDateTime;
import java.util.List;

import org.hibernate.annotations.CreationTimestamp;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Clinic {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer clinicId;

    @Column(nullable = false, columnDefinition = "NVARCHAR(255)")
    private String name;

    @Column(nullable = false, columnDefinition = "NVARCHAR(255)")
    private String address;

    @Column(unique = true, columnDefinition = "NVARCHAR(255)")
    private String phone;

    @Column(columnDefinition = "NVARCHAR(MAX)")
    private String description;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "clinic",cascade = CascadeType.ALL,orphanRemoval = true, fetch = FetchType.LAZY)
    private List<Review> reviews;

    @OneToMany(mappedBy = "clinic",cascade = CascadeType.ALL,orphanRemoval = true, fetch = FetchType.LAZY)
    private List<ClinicImage> clinicImages;
}
